applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_explicit_dynamic_solver.py

```diff
@@ -0,0 +1,101 @@
+
+        self.print_on_rank_zero("::[ExplicitMechanicalSolver]:: Variables ADDED")
+    
+    def AddDofs(self):
+        super(ExplicitMechanicalSolver, self).AddDofs()
+        self._add_dynamic_dofs()
+        KratosMultiphysics.VariableUtils().AddDof(StructuralMechanicsApplication.MIDDLE_VELOCITY_X,self.main_model_part)
+        KratosMultiphysics.VariableUtils().AddDof(StructuralMechanicsApplication.MIDDLE_VELOCITY_Y,self.main_model_part)
+        KratosMultiphysics.VariableUtils().AddDof(StructuralMechanicsApplication.MIDDLE_VELOCITY_Z,self.main_model_part)
+
+        if (self.settings["rotation_dofs"].GetBool()):
+            KratosMultiphysics.VariableUtils().AddDof(StructuralMechanicsApplication.MIDDLE_ANGULAR_VELOCITY_X,self.main_model_part)
+            KratosMultiphysics.VariableUtils().AddDof(StructuralMechanicsApplication.MIDDLE_ANGULAR_VELOCITY_Y,self.main_model_part)
+            KratosMultiphysics.VariableUtils().AddDof(StructuralMechanicsApplication.MIDDLE_ANGULAR_VELOCITY_Z,self.main_model_part)
+
+
+        self.print_on_rank_zero("::[ExplicitMechanicalSolver]:: DOF's ADDED")
+
+
+    #### Specific internal functions ####
+    def _create_solution_scheme(self):
+        scheme_type = self.dynamic_settings["scheme_type"].GetString()
+
+        if(scheme_type == "central_differences"):
+            mechanical_scheme = StructuralMechanicsApplication.ExplicitCentralDifferencesScheme(self.dynamic_settings["max_delta_time"].GetDouble(), 
+                                                                             self.dynamic_settings["fraction_delta_time"].GetDouble(), 
+                                                                             self.dynamic_settings["time_step_prediction_level"].GetDouble(), 
+                                                                             self.dynamic_settings["rayleigh_damping"].GetBool())                                                                     
+        else:
+            err_msg =  "The requested scheme type \"" + scheme_type + "\" is not available!\n"
+            err_msg += "Available options are: \"central_differences\""
+            raise Exception(err_msg)
+        return mechanical_scheme
+
+    def _create_mechanical_solver(self):
+        computing_model_part = self.GetComputingModelPart()
+        mechanical_scheme = self.get_solution_scheme()
+        linear_solver = self.get_linear_solver()
+
+        mechanical_solver = StructuralMechanicsApplication.ExplicitStrategy(computing_model_part,
+                                            mechanical_scheme, 
+                                            linear_solver, 
+                                            self.settings["compute_reactions"].GetBool(), 
+                                            self.settings["reform_dofs_at_each_step"].GetBool(), 
+                                            self.settings["move_mesh_flag"].GetBool())
+
+        mechanical_solver.SetRebuildLevel(1)   
```

Comment:
So you rebuild in every iteration?

Reply:
this was artificial from a previous version. fixed it now